modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2SaasAppTokenRevocationTestCase.java

```diff
@@ -0,0 +1,112 @@
+package org.wso2.identity.integration.test.oauth2;
+
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.context.ConfigurationContextFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
+import org.wso2.carbon.identity.oauth.stub.dto.OAuthConsumerAppDTO;
+import org.wso2.carbon.identity.oauth2.stub.dto.OAuth2TokenValidationRequestDTO;
+import org.wso2.carbon.identity.oauth2.stub.dto.OAuth2TokenValidationRequestDTO_OAuth2AccessToken;
+import org.wso2.carbon.identity.oauth2.stub.dto.OAuth2TokenValidationResponseDTO;
+import org.wso2.identity.integration.common.clients.application.mgt.ApplicationManagementServiceClient;
+import org.wso2.identity.integration.common.clients.oauth.Oauth2TokenValidationClient;
+
+/**
+ * Test class to test token validation after service provider is SaaS-disabled.
+ */
+public class OAuth2SaasAppTokenRevocationTestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+    private static final String TOKEN_API_ENDPOINT = "https://localhost:9853/oauth2/token";
+    private static final String ADMIN_USERNAME = "admin";
+    private static final String ADMIN_PASSWORD = "admin";
+    private static final String TENANT_ADMIN_USERNAME = "admin@wso2.com";
```

Comment:
You can easily get the username & password using automation.xml file. That is the standard way of getting preventing hard-code them inside the class.